minio/parsers.py

```diff
@@ -141,6 +141,39 @@ def parse_list_buckets(data):
         raise InvalidXMLError('Invalid XML provided for "ListBucketsResult" '
                               'some fields are missing. Message: {0}'.format(error.message))
 
+def _parse_objects_from_xml_elts(bucket_name, contents, common_prefixes):
+    """Internal function that extracts objects and common prefixes from
+    list_objects responses.
+    """
+    objects = []
+    for content in contents:
+        key_elt = content.find('s3:Key', _S3_NS)
```

Comment:
Couple of thoughts.

1. cElementTree works with xpath ie `name = tree.find('volStatus/volumes/volume/volName').text`.

2. you could extend or have generic functions on top of  cElementTree classes to aware of namespace and returning urldecoded value by default.  You could avoid repeation of logic like `content.find("element", _S3_NS)`